### PR TITLE
Use URI2 as key for docs

### PR DESCRIPTION
**Pull request by davidanthoff**
The implementation of ``URI2`` now is case-insensitive on Windows, and case-sensitive everywhere else.

This fixes things on Windows, where julia ``include`` statements are case-insensitive.

### CODE REVIEW

File: src/provider_misc.jl [hunk 1]
```diff
@@ -311,7 +311,9 @@ end
 
 
 function process(r::JSONRPC.Request{Val{Symbol("julia/toggle-lint")},TextDocumentIdentifier}, server)
-    server.documents[r.uri]._runlinter != server.documents[r.uri]._runlinter
+    # TODO Isn't this always false? I think this is meant to toggle things
+    # i.e. use assignment rather than a comparison here, right?
+    server.documents[filepath_from_uri(r.uri)]._runlinter != server.documents[filepath_from_uri(r.uri)]._runlinter
```
Comment (davidanthoff) [inline src/provider_misc.jl]: @ZacLN This is just a bug, right? It should simply flip the value of ``_runlinter``?

Comment (ZacLN) [reply src/provider_misc.jl]: yup def a bug

Comment (davidanthoff) [reply src/provider_misc.jl]: Alright, I fixed this with a commit, if you could take a brief look at that as well, it would be great. Is part of this PR.

**Comment by davidanthoff** [comment]
Ok, just realized this doesn't work... We can't assume all URIs are file paths, so we'll have to keep using URIs as the key... I have one other idea for this, I'll try that next. Essentially a URI type that compares case-insensitive for files on win.

**Comment by davidanthoff** [comment]
Alright, this should work now and be ready for review.

I'm now using a new type called ``URI`` (to distinguish it from the type in URIParser.jl), that provides the case-insensitive comparison and hashing for ``file://`` URIs on Windows.

**Pull request merged**